src/Framework/FileClassifier.cs

```diff
@@ -193,17 +193,98 @@ public void RegisterImmutableDirectory(string? directory)
             }
         }
 
-        private static string EnsureTrailingSlash(string fileSpec)
+        public void RegisterFrameworkLocations(Func<string, string?> getPropertyValue)
+        {
+            // Register toolset paths into list of immutable directories
```

Comment:
It seems to be implemented too
https://github.com/dotnet/msbuild/blob/ecabc58f6072cb6d336ea47e63af55c71a2f25d9/src/Build/Definition/ToolsetReader.cs#L402

Can you benefit from that?